sqrt.go

```diff
@@ -0,0 +1,38 @@
+package exact
+
+// Sqrt computes the square root of x.
+// The precision is DefPrecision.
+// To use arbitrary precision, see Sqrtp.
+// Note: this function is orders of magnitude slow than math.Sqrt,
+// but the result can be made as precise as you wish by using
+// Sqrtp.
+func Sqrt(x Frac) Frac {
+	return Sqrtp(x, DefPrecision)
+}
+
+// Sqrtp computes the square root of x using precision prec.
+// Note: Depending on the value of prec, this function can
+// take hours, days, years, a life time, ..., to complete.
+func Sqrtp(x, prec Frac) Frac {
+	if x.P.Cmp(zero) == 0 {
+		return Zero()
+	}
+
+	g := One() // first guess
+
+	for !closeEnough(Div(x, g), g, prec) {
```

Comment:
If you stored Div(x,g) result you could reuse it on the betterGuess function. Since heron's method can have a lot of iterations it will involve half the duplications, perhaps it is worth it =)

Reply:
yes, I thought a moment about this and think that this could obfuscate the simpler idea of making a new guess based on the last one.  But I'll bench this change, maybe makes sense.

Reply:
Bench results for exact.Sqrt using precision 1.0e-100:
```
BenchmarkExactSqrt2-8                     100000             16305 ns/op
BenchmarkExactSqrt2Optimized-8            100000             14994 ns/op
BenchmarkMathSqrt2-8                    2000000000               0.28 ns/op
```
I don't think worth, the code is understandable that way (the idea of approximation is clear).